### PR TITLE
[JENKINS-62421] Added aria-labels to username & password input fields

### PR DESCRIPTION
**Pull request by romenrg**
Since we want to keep the clean aspect while improving accessibility, I opted for aria-labels instead of labels for the username and password input fields in the login form.

See [JENKINS-62421](https://issues.jenkins-ci.org/browse/JENKINS-62421).

### Proposed changelog entries

* Added aria-labels for username and password fields in login form to improve accessibility

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [X] (If applicable) Jira issue is well described
- [X] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [X] Appropriate autotests or explanation to why this change has no tests
- [N/A] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@oleg-nenashev 
@scherler 
@fqueiruga 
@timja 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Comment by romenrg** [comment]
For reference, see the [corresponding section in the Mozilla docs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-label_attribute):

> The aria-label attribute is used to define a string that labels the current element. Use it in cases where a text label is not visible on the screen. If there is visible text labeling the element, use aria-labelledby instead.

File: core/src/main/resources/jenkins/model/Jenkins/login.jelly [hunk 1]
```diff
@@ -83,6 +83,7 @@ THE SOFTWARE.
                 </j:if>
                 <div class="formRow">
                   <input
+                    aria-label="Username"
```
Comment (daniel-beck) [inline core/src/main/resources/jenkins/model/Jenkins/login.jelly:86]:
```suggestion
                    aria-label="${%Username}"
```

This makes the `aria-label` localizable, and actually reuses translations for the placeholder.

Not sure whether the latter is a good idea though.

File: core/src/main/resources/jenkins/model/Jenkins/login.jelly [hunk 2]
```diff
@@ -94,6 +95,7 @@ THE SOFTWARE.
                 </div>
                 <div class="formRow">
                   <input
+                    aria-label="Password"
```
Comment (daniel-beck) [inline core/src/main/resources/jenkins/model/Jenkins/login.jelly:98]:
```suggestion
                    aria-label="${%Password}"
```

As above.

**Review by fqueiruga — APPROVED** [review]

**Comment by romenrg** [comment]
Great point @daniel-beck . I just added a commit with your suggestion of making labels localizable. This way we allow screen readers to read the page in the persons language (provided we have a translation for that property in the given language)

**Review by timja — APPROVED** [review]

**Review by res0nance — APPROVED** [review]

**Review by daniel-beck — APPROVED** [review]
Looks reasonable. All I know about this kind of attribute is from the linked Mozilla page, so this is a very weak +1.

**Review by oleg-nenashev — APPROVED** [review]
Looks very reasonable according to the docs. Thanks @romenrg !

I plan to merge it in 24 hours if no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information

**Pull request merged**